src/qt/optionsmodel.cpp

```diff
@@ -71,9 +72,16 @@ void OptionsModel::Init(bool resetSettings)
     fMinimizeOnClose = settings.value("fMinimizeOnClose").toBool();
 
     // Display
-    if (!settings.contains("nDisplayUnit"))
-        settings.setValue("nDisplayUnit", BitcoinUnits::BTC);
-    nDisplayUnit = settings.value("nDisplayUnit").toInt();
+    if (!settings.contains("DisplayBitcoinUnit")) {
+        settings.setValue("DisplayBitcoinUnit", QVariant::fromValue(BitcoinUnit::BTC));
+    }
+    QVariant unit = settings.value("DisplayBitcoinUnit");
```

Comment:
75832fdc37ea3fe9cf515bd1946e220fe07a440b this is just memoizing the value, so why not
```suggestion
    const QVariant& unit = settings.value("DisplayBitcoinUnit");
```